src/hooks/useSchemaValidator.ts

```diff
@@ -0,0 +1,218 @@
+import { useDatatype } from './useDatatype';
+import { useEffect, useMemo, useState } from 'react';
+import { useConfig } from '@openmrs/esm-framework';
+import { ConceptTrue, ConceptFalse } from '../constants';
+import { OHRIFormField, OHRIFormSchema } from '../api/types';
+
+export function useSchemaValidator(schema: OHRIFormSchema) {
+  //initializing state
+  const [errors, setErrors] = useState([]);
+  const [warnings, setWarnings] = useState([]);
+  const [fullArray, setFullArray] = useState<Array<OHRIFormField>>([]);
```

Comment:
Can you rename this variable to clearly reflect that it's holding fields?

Reply:
Noted.